### PR TITLE
Release v5.10.0

### PR DESCRIPTION
**Pull request by qrkourier**

### CODE REVIEW

**Review by qrkourier — COMMENTED** [review]
pending completion of manual regression testing with Ansible Collection

File: netfoundry/ctl.py [hunk 1]
```diff
@@ -700,7 +709,55 @@ def list(cli, spinner: object = None):
...
+                            resource, status = get_generic_resource_by_type_and_id(setup=organization, resource_type=type_by_prop[k], resource_id=v)
+                            if resource.get('name'):                # if the name property isn't empty
+                                match[k] = f"{resource['name']}"    # wedge the name into the ID column
+
+    # if echo=False then return the object and parent objects instead of printing with an output format
+    if not echo:
+        return filtered_matches, organization
+
     if cli.args.output == "text":
+        if cli.config.list.names:
```
Comment (tmousawNF) [inline netfoundry/ctl.py:739]: Is 739-759 identical to 712-732?

File: netfoundry/ctl.py [hunk 2]
```diff
@@ -898,7 +955,7 @@ def demo(cli):
                 name=network_name,
                 size=cli.config.demo.size,
                 version=cli.config.demo.product_version,
-                wait=0)  # FIXME: don't use wait > 0 until process-executions beta is launched, until then poll for status
+                wait=333)
```
Comment (tmousawNF) [inline netfoundry/ctl.py:958]: Might be better to remove the wait and go with the default from the library (which you thought was 10 minutes).

File: netfoundry/ctl.py [hunk 3]
```diff
@@ -1153,7 +1210,7 @@ def use_organization(cli, spinner: object = None, prompt: bool = True):
             raise NFAPINoCredentials()
     spinner.succeed(f"Logged in profile '{cli.config.general.profile}'")
     cli.log.debug(f"logged-in organization label is {organization.label}.")
-    networks = Networks(Organization=organization)
+    networks = Networks(setup=organization)
```
Comment (tmousawNF) [inline netfoundry/ctl.py:1213]: Left off here on 6/23.

File: netfoundry/network_group.py [hunk 4]
```diff
@@ -176,9 +159,9 @@ def create_network(self, name: str, network_group_id: str = None, location: str
         :param version: optional product version string like 7.3.17
         :param size: optional network configuration metadata name from /core/v2/network-configs e.g. "medium"
         """
-        my_nc_data_centers_by_location = self.nc_data_centers_by_location()
-        if not my_nc_data_centers_by_location.get(location):
-            raise RuntimeError(f"unexpected network location '{location}'. Valid locations include: {', '.join(my_nc_data_centers_by_location.keys())}.")
+        # my_nc_data_centers_by_location = self.nc_data_centers_by_location()
```
Comment (tmousawNF) [inline netfoundry/network_group.py:162]: Should we just remove these three lines?

File: netfoundry/organization.py [hunk 5]
```diff
@@ -9,10 +9,9 @@
 from pathlib import Path
```
Comment (tmousawNF) [inline netfoundry/organization.py:9]: Left off here - 6/27/2022

File: netfoundry/utility.py [hunk 6]
```diff
@@ -1,33 +1,37 @@
...
 import os
-from stat import filemode
 import re  # regex
-from urllib.parse import urlparse
 import time  # enforce a timeout; sleep
 import unicodedata  # case insensitive compare in Utility
 from dataclasses import dataclass, field
-from re import sub
+from json import JSONDecodeError
+# from re import sub
```
Comment (tmousawNF) [inline netfoundry/utility.py:11]: Should this line just be deleted?

File: netfoundry/utility.py [hunk 7]
```diff
@@ -215,32 +235,117 @@ def is_uuidv4(string: str):
         return True
 
 
-def eprint(*args, **kwargs):
-    """Adapt legacy function to logging."""
-    logging.debug(*args, **kwargs)
+# def eprint(*args, **kwargs):
```
Comment (tmousawNF) [inline netfoundry/utility.py:238]: Should this function just be removed?

**Review by tmousawNF — APPROVED** [review]
Approving with the understanding that you'll address the comments.

**Pull request merged**